TutorLizard.Web/Controllers/AccountController.cs

```diff
@@ -23,15 +27,54 @@ public IActionResult Index()
         return View();
     }
 
-    public IActionResult Login([FromQuery] string? returnUrl)
+    public async Task<IActionResult> Login([FromQuery] string? returnUrl)
     {
         if (returnUrl is not null)
         {
             TempData["returnUrl"] = returnUrl;
         }
+
         return View();
     }
 
+    public async Task GoogleLogin()
+    {
+        await HttpContext.ChallengeAsync(GoogleDefaults.AuthenticationScheme,
+            new AuthenticationProperties
+            {
+                RedirectUri = Url.Action("GoogleResponse"),
+                Items =
+                {
+                    { "prompt", "select_account" }
+                }
+            });
+    }
+
+    public async Task<IActionResult> GoogleResponse()
+    {
+        var result = await HttpContext.AuthenticateAsync(GoogleDefaults.AuthenticationScheme);
+
+        if (result?.Succeeded != true)
+        {
+            return RedirectToAction("Login");
+        }
+
+        var claims = result.Principal.Identities.FirstOrDefault()?.Claims.ToList();
+
+        var claimNameIdentifier = claims?.FirstOrDefault(x => x.Type == ClaimTypes.NameIdentifier)?.Value;
+        var claimEmail = claims?.FirstOrDefault(x => x.Type == ClaimTypes.Email)?.Value;
+
+        
+        var loggedIn = await _userAuthenticationService.LogInAsync(claimNameIdentifier, claimEmail);
```

Comment:
You provide email as password to `LoginAsync`, so it should almost never actually work. However, if you put breakpoint at the very begining of this action, you can see that user claims from Google are already filled in.

When I use this, I get a result of being logged in as my Google account, but I get redirected back to login page. I can access pages for logged in user, but I can't do anything, because my requests bounce back on `userId is null` checks.

I think a way to handle it could be to read what we need from the claims (I'm not sure if `AuthenticateAsync` is needed, claims can be found in `User`), clear them (log out, essentially) and then try to log in using what you read, but for that, we'll need a different method than `LogInAsync`, because we have different input.

I don't think there's a way right now to actually log in, because there's no way to register using Google account, so we cannot know what the correct `NameIdentifier` we get from Google should be (in my case it's a string of numbers). If we saved this to db during registration, we could use it combined with email as a way to log in.
Also, I don't know if it always should be a string of numbers (like some Google-given Id to verify it's the same account as before) or if it could be a nickname if some registration process is put in place. We can use it, either way.